test/unit/host_test.rb

```diff
@@ -2137,6 +2137,13 @@ def to_managed!
     assert_equal 'foo', host.shortname
   end
 
+  test 'check operatingsystem and architecture association' do
+    primary = FactoryGirl.build(:nic_managed, :primary => true, :name => 'foo')
+    host = FactoryGirl.build(:host, :interfaces => [primary, FactoryGirl.build(:nic_managed, :provision => true)])
```

Comment:
You can probably use just `:host` without creating NICs explicitly.  Worst case, use `:host, :managed` or `:host, :interfaces => [FactoryGirl.build(:nic_primary_and_provision)]`.